name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/columns/DividendPaymentColumn.java

```diff
@@ -0,0 +1,170 @@
+package name.abuchen.portfolio.ui.views.columns;
+
+import java.time.LocalDate;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.eclipse.jface.viewers.ColumnLabelProvider;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+
+import name.abuchen.portfolio.model.Adaptor;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.SecurityEvent.DividendEvent;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.ui.Messages;
+import name.abuchen.portfolio.ui.util.Colors;
+import name.abuchen.portfolio.ui.util.viewers.Column;
+import name.abuchen.portfolio.ui.util.viewers.ColumnViewerSorter;
+import name.abuchen.portfolio.ui.util.viewers.DateLabelProvider;
+
+public class DividendPaymentColumn
+{
+
+    static Column createNextDividendExDateColumn()
+    {
+        Column column = new Column("nexdd", Messages.ColumnDividendsNextExDate, SWT.LEFT, 80); //$NON-NLS-1$
+        column.setMenuLabel(Messages.ColumnDividendsNextExDate_MenuLabel);
+        column.setGroupLabel(Messages.GroupLabelDividends);
+        column.setVisible(false);
+
+        LocalDate now = LocalDate.now();
+
+        Function<Object, LocalDate> dataProvider = element -> {
+            Security sec = Adaptor.adapt(Security.class, element);
+            if (sec == null)
+            {
+                return null; //
+            }
+            return sec.getEvents().stream() //
+                            .filter(DividendEvent.class::isInstance) //
+                            .map(e -> ((DividendEvent) e).getDate()) //
+                            .sorted(Comparable::compareTo) //
+                            .filter(d -> d.isAfter(now)) //
+                            .findFirst().orElse(null);
+        };
```

Comment:
Maybe this is a more elegant solution?
```suggestion
LocalDate now = LocalDate.now();

Function<Object, LocalDate> dataProvider = element -> {
    Security security = Adaptor.adapt(Security.class, element);
    return Optional.ofNullable(security)
                   .map(s -> s.getEvents().stream()
                               .filter(DividendEvent.class::isInstance)
                               .map(e -> ((DividendEvent) e).getDate())
                               .filter(d -> d.isAfter(now))
                               .min(Comparable::compareTo)
                               .orElse(null))
                   .orElse(null);
};
```

Reply:
"Elegant" is a quite subjective term ;-) When writing code my goal is to produce left-documenting code and - this is biased of course - but I think that my version is explaining its logic more clearly to a reader: Take all DividendEvents, sort their ex-dates in an ascending order, throw away all dates that are in the past and take the first that remains.

Your version IMHO doesn't provide that and is a bit confusing with `orElse` that is needed to be used for the stream and the Optional. So TBH: I'd prefer to keep my version.